CalibTracker/SiPixelESProducers/interface/SiPixelGainCalibrationForHLTGPU.h

```diff
@@ -0,0 +1,32 @@
+#ifndef CalibTracker_SiPixelESProducers_interface_SiPixelGainCalibrationForHLTGPU_h
+#define CalibTracker_SiPixelESProducers_interface_SiPixelGainCalibrationForHLTGPU_h
+
+#include "CondFormats/SiPixelObjects/interface/SiPixelGainCalibrationForHLT.h"
+#include "HeterogeneousCore/CUDACore/interface/ESProduct.h"
+
+class SiPixelGainCalibrationForHLT;
+class SiPixelGainForHLTonGPU;
+struct SiPixelGainForHLTonGPU_DecodingStructure;
+class TrackerGeometry;
+
+class SiPixelGainCalibrationForHLTGPU {
+public:
+  explicit SiPixelGainCalibrationForHLTGPU(const SiPixelGainCalibrationForHLT &gains, const TrackerGeometry &geom);
+  ~SiPixelGainCalibrationForHLTGPU();
+
+  const SiPixelGainForHLTonGPU *getGPUProductAsync(cudaStream_t cudaStream) const;
+  const SiPixelGainForHLTonGPU *getCPUProduct() const { return gainForHLTonHost_; }
+  const SiPixelGainCalibrationForHLT *getOriginalProduct() { return gains_; }
```

Comment:
Even considering the coding rules, I have a preference for `getGPUProductAsync()` vs `gpuProductAsync()`, because "Async" applies to getting the product, not to the product itself.

However, I can make the change if people insist.

Reply:
Thaks for clarifying. I agree for `getGPUProductAsync`, since it also more clearly indicates some action and is not really falling under 2.11 as direct member access.

The other two fall under 2.11 as direct member access.

Reply:
Fixed in cms-patatrack#601